Streamer test flake

This happens infrequently. Keeping logs here as a reference in case someone wants to tackle this.

```
=== RUN   TestStreamer_Stream/in_general
    streamer_test.go:55: 
        	Error Trace:	streamer_test.go:55
        	            				asm_amd64.s:1374
        	Error:      	Not equal: 
        	            	expected: event.Event{Type:"event.octant.dev/app-logs", Data:[]log.Message{log.Message{ID:"1", Date:0, LogLevel:"", Location:"", Text:"", JSON:"", StackTrace:""}}, Err:error(nil)}
        	            	actual  : event.Event{Type:"event.octant.dev/app-logs", Data:[]log.Message(nil), Err:error(nil)}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,13 +2,3 @@
        	            	  Type: (event.EventType) (len=25) "event.octant.dev/app-logs",
        	            	- Data: ([]log.Message) (len=1) {
        	            	-  (log.Message) {
        	            	-   ID: (string) (len=1) "1",
        	            	-   Date: (int64) 0,
        	            	-   LogLevel: (string) "",
        	            	-   Location: (string) "",
        	            	-   Text: (string) "",
        	            	-   JSON: (string) "",
        	            	-   StackTrace: (string) ""
        	            	-  }
        	            	- },
        	            	+ Data: ([]log.Message) <nil>,
        	            	  Err: (error) <nil>
        	Test:       	TestStreamer_Stream/in_general
panic: test timed out after 10m0s

goroutine 43 [running]:
testing.(*M).startAlarm.func1()
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1628 +0xe5
created by time.goFunc
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/time/sleep.go:167 +0x45

goroutine 1 [chan receive]:
testing.(*T).Run(0xc0003a9080, 0x1e3c7dc, 0x13, 0x1ec85e0, 0x1092a01)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1179 +0x3ad
testing.runTests.func1(0xc000303380)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1449 +0x78
testing.tRunner(0xc000303380, 0xc00020bde0)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1127 +0xef
testing.runTests(0xc000376400, 0x2697d80, 0x8, 0x8, 0xbfdbdd2b1341ac18, 0x8bb3288b01, 0x26a7ec0, 0x100f9f0)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1447 +0x2e8
testing.(*M).Run(0xc000201b80, 0x0)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1357 +0x245
main.main()
	_testmain.go:57 +0x138

goroutine 19 [chan receive]:
k8s.io/klog/v2.(*loggingT).flushDaemon(0x26a8140)
	/Users/runner/work/octant/octant/vendor/k8s.io/klog/v2/klog.go:1131 +0x8b
created by k8s.io/klog/v2.init.0
	/Users/runner/work/octant/octant/vendor/k8s.io/klog/v2/klog.go:416 +0xd8

goroutine 38 [chan receive]:
testing.(*T).Run(0xc0003a9200, 0x1e34ea5, 0xa, 0x1ec85d8, 0xc00005d770)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1179 +0x3ad
github.com/vmware-tanzu/octant/internal/log.TestStreamer_Stream(0xc0003a9080)
	/Users/runner/work/octant/octant/internal/log/streamer_test.go:34 +0x8b
testing.tRunner(0xc0003a9080, 0x1ec85e0)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1127 +0xef
created by testing.(*T).Run
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1178 +0x386

goroutine 37 [chan receive]:
github.com/vmware-tanzu/octant/internal/log.NewStreamer.func1(0xc000372d20, 0xc00033cc00)
	/Users/runner/work/octant/octant/internal/log/streamer.go:59 +0x156
created by github.com/vmware-tanzu/octant/internal/log.NewStreamer
	/Users/runner/work/octant/octant/internal/log/streamer.go:58 +0x106

goroutine 39 [chan receive]:
github.com/vmware-tanzu/octant/internal/log.TestStreamer_Stream.func1(0xc0003a9200)
	/Users/runner/work/octant/octant/internal/log/streamer_test.go:61 +0x22e
testing.tRunner(0xc0003a9200, 0x1ec85d8)
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1127 +0xef
created by testing.(*T).Run
	/Users/runner/hostedtoolcache/go/1.15.2/x64/src/testing/testing.go:1178 +0x386

goroutine 40 [chan receive]:
github.com/vmware-tanzu/octant/internal/log.NewStreamer.func1(0xc000372d80, 0xc00033cc40)
	/Users/runner/work/octant/octant/internal/log/streamer.go:59 +0x156
created by github.com/vmware-tanzu/octant/internal/log.NewStreamer
	/Users/runner/work/octant/octant/internal/log/streamer.go:58 +0x106
FAIL	github.com/vmware-tanzu/octant/internal/log	600.388s
?   	github.com/vmware-tanzu/octant/internal/mime	[no test files]
```

- Octant version (use `octant version`): 0.16.1

--- a/changelogs/unreleased/1797-GuessWhoSamFoo
+++ b/changelogs/unreleased/1797-GuessWhoSamFoo
@@ -1,0 +1,1 @@
+Fix streamer test flake

--- a/internal/log/streamer.go
+++ b/internal/log/streamer.go
@@ -122,7 +122,6 @@ func (lm *Streamer) Stream(readyCh <-chan struct{}) (<-chan event.Event, func())
 		}
 
 		<-readyCh
-		ch <- lm.createEvent()
 	}()
 
 	lm.listeners[id] = ch

--- a/internal/log/streamer_test.go
+++ b/internal/log/streamer_test.go
@@ -36,6 +36,12 @@ func TestStreamer_Stream(t *testing.T) {
 				ID: "1",
 			}
 
+			want := event.Event{
+				Type: event.EventTypeAppLogs,
+				Data: []Message{message},
+				Err:  nil,
+			}
+
 			f := newStubMessageListenerFactory()
 			s := NewStreamer(f)
 
@@ -45,20 +51,16 @@ func TestStreamer_Stream(t *testing.T) {
 			ch, cancel := s.Stream(readyCh)
 
 			done := make(chan struct{}, 1)
+			var e event.Event
 			go func() {
-				e := <-ch
-				want := event.Event{
-					Type: event.EventTypeAppLogs,
-					Data: []Message{message},
-					Err:  nil,
-				}
-				require.Equal(t, want, e)
-				close(done)
+				defer close(done)
+				e = <-ch
 			}()
 
 			f.ch <- message
 
 			<-done
+			require.Equal(t, want, e)
 			cancel()
 
 		})

